src/System.Linq.Parallel/src/System/Linq/ParallelEnumerable.cs

```diff
@@ -5354,18 +5351,20 @@ public static ParallelQuery<TResult> OfType<TResult>(this ParallelQuery source)
         }
 
         /// <summary>
-        /// Converts the elements of a ParallelQuery to the specified type.
+        /// Converts the elements of a weakly-typed ParallelQuery to the specified stronger type.
         /// </summary>
-        /// <typeparam name="TResult">The type to convert the elements of <paramref name="source"/> to.</typeparam>
-        /// <param name="source">The sequence that contains the elements to be converted.</param>
+        /// <typeparam name="TResult">The stronger type to convert the elements of <paramref name="source"/> to.</typeparam>
+        /// <param name="source">The sequence that contains the weakly typed elements to be converted.</param>
         /// <returns>
-        /// A sequence that contains each element of the source sequence converted to the specified type.
+        /// A sequence that contains each weakly-type element of the source sequence converted to the specified stronger type.
         /// </returns>
         /// <exception cref="T:System.ArgumentNullException">
         /// <paramref name="source"/> is a null reference (Nothing in Visual Basic).
         /// </exception>
         public static ParallelQuery<TResult> Cast<TResult>(this ParallelQuery source)
         {
+            if (source == null) throw new ArgumentNullException("source");
```

Comment:
> Isn't the desktop stuff eventually pulling these changes in anyways?

For perf improvements and the like, hopefully. But breaking changes are something that need to be discussed in detail.

I'd suggest not making this change here; it should be done as part of a separate PR if we want to make the change.